doc/application/application.rst

```diff
@@ -966,50 +966,60 @@ preferred, since it correctly handles dependencies between options.
 Device Tree Overlays
 ====================
 
-Zephyr uses Device Tree to describe the hardware it runs on. Sometimes it is
-necessary to modify the definitions in the :file:`.dts` files provided in the
-board folder in order to customize settings for a particular application.
-For more information on Device Tree see :ref:`device-tree`.
+As described in :ref:`device-tree`, Zephyr uses Device Tree to
+describe the hardware it runs on. This section describes how you can
+modify an application build's device tree using overlay files.
 
-The Zephyr build system takes a Device Tree configuration
-option's value from the first source in which it is specified, taken from the
-following available sources, in order:
+Overlay files, which customarily have the :file:`.overlay` extension,
+contain device tree fragments which add to or modify the device tree
+used while building a Zephyr application. To add an overlay file or
+files to the build, set the CMake variable :makevar:`DTC_OVERLAY_FILE`
+to a whitespace-separated list of your overlay files.
 
-#. An application's current DT configuration (i.e. the file
-   :file:`zephyr/<board>.dts.pre.tmp` in the build directory).
+The Zephyr build system begins creation of a device tree by running
+the C preprocessor on a file which includes the following:
```

Comment:
You've inverted the order here which makes it inconsistent with the equivlalent list describing the Kconfig and `.conf` processing. I would recommend we keep both in sync in terms of order?

Reply:
Do we really need to tell the user that we generate that file and then dtc is run on it? sounds like an implementation detail

Reply:
> You've inverted the order here which makes it inconsistent with the equivlalent list describing the Kconfig and .conf processing. I would recommend we keep both in sync in terms of order?

What do you mean? The new text is a description of how the DT is built; the old text is just incorrect AFAICT. What am I missing?

> Do we really need to tell the user that we generate that file and then dtc is run on it? sounds like an implementation detail

The reason I chose to include it is because it lets them know that they can use preprocessor symbols -- and in particular, which ones. That should help them write their overlays.

I can remove it if you prefer this to remain undocumented.